packages/fxa-graphql-api/src/backend/legal.service.spec.ts

```diff
@@ -0,0 +1,113 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Provider } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ConfigService } from '@nestjs/config';
+import { LegalService } from './legal.service';
+
+async function fetch(url: string) {
+  if (
+    url.includes('/settings/legal-docs/firefox_cloud_services_tos_locales.json')
+  ) {
+    return {
+      ok: true,
+      json: () => ['en', 'es'],
+    };
+  }
+
+  if (url.includes('en/firefox_cloud_services_tos')) {
+    return {
+      ok: true,
+      text: () => '# Firefox Cloud Services: Terms of Service',
+    };
+  }
+  if (url.includes('es/firefox_cloud_services_tos')) {
+    return {
+      ok: true,
+      text: () => '# Firefox Cloud Services: Condiciones del servicio',
+    };
+  }
+
+  return {
+    ok: true,
+    text: () => '',
+  };
+}
+
+jest.mock('node-fetch', () => fetch);
+
+describe.only('#unit - LegalService', () => {
+  let service: LegalService;
+
+  beforeEach(async () => {
+    jest.resetModules();
+
+    const MockConfig: Provider = {
+      provide: ConfigService,
+      useValue: {
+        get: jest.fn().mockReturnValue('http://localhost:3030'),
+      },
+    };
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [MockConfig, LegalService],
+    }).compile();
+
+    service = module.get<LegalService>(LegalService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('returns english doc', async () => {
+    const doc = await service.getDoc('en', 'firefox_cloud_services_tos');
+    expect(doc).toBeDefined();
+    expect(doc.markdown).toBeDefined();
+    expect(
+      /# Firefox Cloud Services: Terms of Service/.test(doc.markdown)
+    ).toBeTruthy();
+  });
+
+  it('returns spanish doc', async () => {
+    const doc = await service.getDoc('es-ES', 'firefox_cloud_services_tos');
+
+    expect(doc).toBeDefined();
+    expect(doc.markdown).toBeDefined();
+    expect(
+      /# Firefox Cloud Services: Condiciones del servicio/.test(doc.markdown)
+    ).toBeTruthy();
+  });
+
+  it('returns empty string for bogus doc', async () => {
+    const doc = await service.getDoc('en', 'xyz');
+
+    expect(doc).toBeDefined();
+    expect(doc.markdown).toBeDefined();
+    expect(doc.markdown).toEqual('');
+  });
+
+  describe('invalid files', () => {
+    async function test(fileName: string) {
+      await expect(async () => {
+        await service.getDoc('en', fileName);
+      }).rejects.toThrow('Invalid file name');
+    }
+
+    it('rejects empty file name', async () => {
+      await test('');
+    });
+
+    it('rejects relative path like file name', async () => {
+      await test('../foo.txt');
+    });
+
+    it('rejects absolute path like file name', async () => {
+      await test('/foo/bar.txt');
+    });
+
+    it('rejects long file name', async () => {
+      await test(Array(1000).fill('a').join(''));
```

Comment:
Alternatively, [`"a".repeat(1000)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat) should work.

Also, what is a "long" filename? Are we specifying a maximum [valid] file name length somewhere?

**UPDATE:** Ah, below I see this, so maybe 500 characters is the max valid length:

https://github.com/mozilla/fxa/blob/0644005b3c41f236fadde83f53c14eb305615cae/packages/fxa-graphql-api/src/backend/legal.service.ts#L25-L27